xbmc/application/ApplicationPlayerCallback.cpp

```diff
@@ -92,12 +84,10 @@ void CApplicationPlayerCallback::OnPlayBackStarted(const CFileItem& file)
     CServiceBroker::GetJobManager()->PauseJobs();
   }
 
-  CServiceBroker::GetPVRManager().OnPlaybackStarted(*m_itemCurrentFile);
+  CServiceBroker::GetPVRManager().OnPlaybackStarted(*itemCurrentFile);
```

Comment:
Maybe move to CApplication like the other calls to GetPVRManager for consistency, but don't know if exact timing of following line ```stackHelper->OnPlayBackStarted(file)``` is important for PVR.

Reply:
For me it is logically correct to have this call here. We are in ApplicationPlayCallBack and inform the otehr components from here about playback state changes.

Reply:
Your call. I was only pointing out the inconsistency with the other PVR calls moved to CApplication.

Reply:
@dobo90 i agree to @CrystalP if the other pvr manager calls were moved to `CApplication`, then this one here also should be moved, for consistency.

@CastagnaIT this should be changed before we merge this PR.

Reply:
What about the call stackHelper->OnPlayBackStarted(file)? I'm missing a bit the consistency here as well? Which callbacks are now done in CApplication and why not all of them? @dobo90 how did you decide what needs to stay here and what has to be moved?

Reply:
@ksooo
It moved also `CServiceBroker::GetPVRManager().OnPlaybackStarted`.

Regarding stackHelper: personally I don't like how it's designed but I believe it shouldn't crash the application if it's accessed from different threads. It relies on the fact that CApplicationStackHelper::m_currentStack (CFileItemList) is thread safe which is assured by CFileItemList::m_lock. CApplicationStackHelper also uses separate critical section in CApplicationStackHelper::OnPlayBackStart - probably it should be extended to other methods of that class. But I don't think it should be done as a scope of this PR. This PR makes sure that members of CApplicationPlayerCallback are race free (because they were moved to CApplication).

Regarding what needs to be moved from CApplicationPlayerCallback to CApplication. CVideoPlayer invokes methods from CVideoPlayer::m_callback in different contexts, for example:
```
      IPlayerCallback *cb = &m_callback;
      m_outboundEvents->Submit([=]() {
        cb->OnAVChange();
      });
```
Every method of CApplicationPlayerCallback should be thread safe. The rule of thumb is to get rid of all members of CApplicationPlayerCallback and access them in CApplication. This makes CApplicationPlayerCallback transfer control flow to the main thread.